benefits/core/views.py

```diff
@@ -60,6 +59,28 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class AgencyCardView(RedirectView):
+    """View handler forwards the request to the agency's Agency Card (e.g. Eligibility API) flow, or returns a user error."""
+
+    pattern_name = routes.ELIGIBILITY_CONFIRM
+
+    @method_decorator(pageview_decorator)
+    def get(self, request, *args, **kwargs):
+        # keep a reference to the agency before removing from kwargs
+        # since the eventual reverse() lookup doesn't expect this key in the kwargs for routes.ELIGIBILITY_CONFIRM
+        # self.kwargs still contains the agency if needed
+        agency = kwargs.pop("agency")
+        session.reset(request)
+        session.update(request, agency=agency, origin=agency.index_url)
+
+        eligibility_api_flow = agency.enrollment_flows.exclude(api_request=None).order_by("id").last()
+        if eligibility_api_flow:
+            session.update(self.request, flow=eligibility_api_flow)
```

Comment:
Minor thing, maybe we can use `request` here directly since we have access to it instead of `self.request`?

Reply:
Incoming with the rebase...

Reply:
Good call, I think this was a holdover from when I had originally implemented this logic in `get_redirect_url` (where it didn't make as much sense, hence moving it all to `get`)

[`cd28963`](https://github.com/cal-itp/benefits/pull/3378/commits/cd2896339792ebcfe0319f74b49e60ea7e5c5a8b)